sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,556 @@
+ * ### Adding credentials
+ *
+ * Before using this module, either as a server or a client, we need to first
+ * add the credentials to be used for the encryption using
+ * @ref net_credman "credman". Note that credman does not copies the credentials
+ * given into the system, it only have information about the credentials and where
+ * it is located at. So it is your responsibility to make sure that the credential
+ * is valid during the lifetime of your application.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include <stdio.h>
+ *
+ * #include "net/credman.h"
+ *
+ * static uint8_t psk_key = "secretPSK"
+ * static ecdsa_public_key_t other_pubkeys[] = {
+ *     { .x = client_pubkey_x, .y = client_pubkey_y },
+ * };
+ *
+ * static const unsigned char server_ecdsa_priv_key[] = {...};
+ * static const unsigned char server_ecdsa_pub_key_x[] = {...};
+ * static const unsigned char server_ecdsa_pub_key_y[] = {...};
+ * static const unsigned char client_pubkey_x[] = {...};
+ * static const unsigned char client_pubkey_y[] = {...};
+ *
+ * int main(void)
+ * {
+ *     credman_credential_t psk_credential = {
+ *         .type = CREDMAN_TYPE_PSK,
+ *         .tag = DTLS_SERVER_TAG,
+ *         .params = {
+ *             .psk = {
+ *                 .key = { .s = psk_key, .len = sizeof(psk_key), },
+ *             },
+ *         },
+ *     };
+ *     res = credman_add(&psk_credential);
+ *     if (res < 0) {
+ *         printf("Error cannot add credential to system: %d\n", (int)res);
+ *     }
+
+ *     credman_credential_t credential = {
+ *         .type = CREDMAN_TYPE_ECDSA,
+ *         .tag = DTLS_SOCK_SERVER_TAG,
+ *         .params = {
+ *             .ecdsa = {
+ *                 .private_key = server_ecdsa_priv_key,
+ *                 .public_key = {
+ *                     .x = server_ecdsa_pub_key_x,
+ *                     .y = server_ecdsa_pub_key_y,
+ *                 },
+ *                 .client_keys = other_pubkeys,
+ *                 .client_keys_size = sizeof(other_pubkeys) / sizeof(other_pubkeys[0]),
+ *             },
+ *         },
+ *     };
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Above we see an example how to register a PSK and a ECC credential.
+ *
+ * First we need to include the header file for @ref net_sock_dtls "DTLS sock".
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/credman.h"
+ *
+ * int main(void)
+ * {
+ *     credman_credential_t psk_credential = {
+ *         .type = CREDMAN_TYPE_PSK,
+ *         .tag = DTLS_SERVER_TAG,
+ *         .params = {
+ *             .psk = {
+ *                 .key = { .s = psk_key, .len = sizeof(psk_key), },
+ *             },
+ *         },
+ *     };
+ *
+ *     [...]
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * We tell @ref net_credman "credman" which credential to add by filling in
+ * the credentials information in a struct @ref credman_credential_t. We are
+ * using `CREDMAN_TYPE_PSK` for credman_credential_t::type for PSK credential.
```

Comment:
Added link to `CREDMAN_TYPE_PSK`. Currently, `make doc` will return
```
warning: unable to resolve reference to `CREDMAN_TYPE_PSK' for \ref command`
```
because the enum values are not documented in `credman.h`. I'll open a PR to fix this in credman.

Reply:
Opened PR #11965

Reply:
#11965 was merged so I believe this needs to be rebased.

Reply:
Rebased. `make doc` returns no warning anymore.